googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -367,6 +367,15 @@ enum CallReaction {
 
 }  // namespace internal
 
+template <typename M>
```

Comment:
Then IMO that's pointing to a structural flaw, and forward decls aren't the fix.
Let's use a trait and have the mock wrappers specialize the trait.